### PR TITLE
feat: drive abci custom loggers configuration

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

We need to log the grovedb operations on testnet so we can easily debug sources of non-determinism. Since grovedb operations contain a lot of data we don't want to push it to elastic. We need a separate file to write grovedb operations there. To allow such configuration, Dashmate should provide options to define custom Drive ABCI loggers.

## What was done?

- Allow to configure any number of loggers in Dashmate.
- Accept a logging specification string (`RUST_LOG`) as logger level in Drive ABCI
- Refactored logging module in Drive ABCI

New Drive's logs configuration format:
- `destination` - `stdout`, `stderr` or file path
- `level` - `error`, `warn`, `info`, `debug`, `trace`, `silent` or logging specification string in RUST_LOG format
- `format` - `full`, `compact`, `pretty`, `json`
- `color` - boolean or null (autodetect)

## How Has This Been Tested?

With tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/src/logging/level.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        Ok(level)
+    }
+}
+
+impl From<&LogLevel> for EnvFilter {
+    fn from(value: &LogLevel) -> Self {
+        match value {
+            LogLevel::Silent => EnvFilter::default(),
+            LogLevel::Custom(configuration) => EnvFilter::new(configuration),
```
Comment (lklimek) [inline packages/rs-drive-abci/src/logging/level.rs:96]: Maybe we should use EnvFilter::try_new (and change this impl to TryFrom) to detect errors in user-supplied level spec?

File: packages/dashmate/configs/defaults/getBaseConfigFactory.js [hunk 2]
```diff
@@ -160,17 +160,12 @@ function getBaseConfigFactory(homeDir) {
...
                 level: 'info',
-              },
-              prettyFile: {
-                level: 'silent',
-                path: homeDir.joinPath('logs', 'base', 'drive-pretty.log'),
-              },
-              jsonFile: {
-                level: 'silent',
-                path: homeDir.joinPath('logs', 'base', 'drive-json.log'),
+                format: 'pretty',
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/configs/defaults/getBaseConfigFactory.js:167]: Should I set full or compact?

File: packages/rs-drive-abci/src/logging/level.rs (hunk 1 above)
Comment (shumkov) [reply packages/rs-drive-abci/src/logging/level.rs:96]: Sounds good to me

File: packages/dashmate/configs/defaults/getBaseConfigFactory.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lklimek) [reply packages/dashmate/configs/defaults/getBaseConfigFactory.js:167]: Full, as it's also default for [tracing_subscriber](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/index.html#)

(however, my personal preference is json ;) )

[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/configs/defaults/getBaseConfigFactory.js]: Oh, I set. Compact.

**Review by lklimek — DISMISSED** [review]

**Pull request merged**